### ISSUE TITLE
Colorscheme is run twice on startup with a minimum init

### ISSUE THREAD

**Issue by katawful** [katawful/kat.nvim#36]
With a minimal init.lua that just loads kat.nvim on startup, the colorscheme runs twice. This increases startup time. I do not know where the bug comes from

**Comment by rewhile** [comment]
See `:h background`. Probably because of `o:background` reloading

I don't understand fennel but probably around these lines? https://github.com/katawful/kat.nvim/blob/041732950b387fee332feee2c7ee4e63f119e7a2/fnl/katdotnvim/main.fnl#L30

Also to respect vanilla vim `:set background=light/dark` I added a hacky workaround: https://github.com/catppuccin/nvim/blob/cd676faa020b34e6617398835b5fa3d1c2e8895c/lua/catppuccin/init.lua#L70

**Comment by katawful** [comment]
My understanding with options is that setting a var to it wouldn't run the side effects (change background and colorscheme in this case). Is this wrong?

As well, I don't know why it happens with a minimal init.lua still, it should still happen multiple times with any init I'd think?

**Comment by rewhile** [comment]
Sorry I am on phone but `:h g:colors_name` does explain it pretty well I think:

![Screenshot_2022-11-07-19-06-39-767_com termux](https://user-images.githubusercontent.com/56817415/200306506-afe73107-be13-4009-accb-0bbf1ad2ff7f.jpg)

**Comment by katawful** [comment]
I don't set the background option in my colorscheme. I simply absorb whatever the background option is so that colors look appropriate

**Comment by rewhile** [comment]
Yeah the colorscheme is also reloaded after setting `g:colors_name` hence I have the lock hack: https://github.com/catppuccin/nvim/blob/cd676faa020b34e6617398835b5fa3d1c2e8895c/lua/catppuccin/init.lua#L70

Unfortunately it is an ugly hack in order to respect o:background. A bit late to say but I also didn't investigate why it was called twice.

**Comment by katawful** [comment]
But it only affects a minimal init. It does not happen with my own init

**Comment by rewhile** [comment]
Are you sure?

**Comment by katawful** [comment]
Yes, I noticed it when testing in a minimal environment during a large refactor to make sure my own dotfiles weren't messing anything up

**Comment by rewhile** [comment]
May I have the minimal config you used? I will take a look

**Comment by katawful** [comment]
I don't have access to my computer atm, but it was not much more than packer.nvim loading in my colorscheme

**Comment by katawful** [comment]
Init file i used

```lua

local execute = vim.api.nvim_command
local fn = vim.fn
local fmt = string.format

-- make the package path ~/.local/share/nvim/plug
local packer_path = "/tmp/nvim" .. "/site/pack"

function ensure (user, repo)
	local install_path = fmt("%s/packer/start/%s", packer_path, repo, repo)
	if fn.empty(fn.glob(install_path)) > 0 then
		execute(fmt("!git clone https://github.com/%s/%s %s", user, repo, install_path))
		execute(fmt("packadd %s", repo))
	end
end

ensure("wbthomason", "packer.nvim")

ensure("katawful", "kat.nvim")
vim.cmd("colorscheme kat.nvim")

```
